frontend/src/components/regions/UserActions/utils.js

```diff
@@ -0,0 +1,61 @@
+export function updateBurnDownSnapshots({ burnDownSnapshots }) {
+  const datesArray = [];
+  burnDownSnapshots.forEach((burnDownSnapshot) => {
+    datesArray.push(burnDownSnapshot.timestamp);
+  });
+
+  const uniqueDatesArr = Array.from(new Set(datesArray)).sort(
+    (a, b) => new Date(a) - new Date(b)
+  );
+
+  const firstDate = new Date(new Date(uniqueDatesArr[0]).getTime());
```

Comment:
I wonder if this code will error with an empty snapshot array. If the api call that fetches the data hasn't finished then the array might be empty.

Can you add another test that just makes sure nothing breaks when the array is empty?